api/service/stagedstreamsync/downloader.go

```diff
@@ -68,9 +68,17 @@ func NewDownloader(host p2p.Host, bc core.BlockChain, consensus *consensus.Conse
 
 	ctx, cancel := context.WithCancel(context.Background())
 
+	mainDB, dbs, err := InitDB(ctx, bc.ShardID(), config, isBeaconNode, dbDir)
+	if err != nil {
+		logger.Error().Err(err).Msg("failed to init db")
+		cancel()
+		return nil
+	}
+
```

Comment:
How this helps? What's the reason to move staged sync db initiation to downloader?

Reply:
No real need, but having db and struct initializations as different steps makes it more clear.

Reply:
Please keep it under staged stream sync. Downloader itself starts staged sync creation. There we initiate db.